internal/handlers/common/aggregations/operators/type.go

```diff
@@ -42,81 +44,88 @@ func newType(args ...any) (Operator, error) {
 		)
 	}
 
-	return &typeOp{
-		param: args[0],
-	}, nil
-}
+	operator := new(typeOp)
 
-// Process implements Operator interface.
-func (t *typeOp) Process(doc *types.Document) (any, error) {
-	typeParam := t.param
+	switch param := args[0].(type) {
+	case *types.Document:
+		if !IsOperator(param) {
+			operator.param = param
+			break
+		}
 
-	var paramEvaluated bool
+		op, err := NewOperator(param)
+		if err != nil {
+			var opErr OperatorError
+			if !errors.As(err, &opErr) {
+				return nil, lazyerrors.Error(err)
+			}
 
-	var res any
+			if opErr.Code() == ErrInvalidExpression {
+				opErr.code = ErrInvalidNestedExpression
+			}
 
-	for !paramEvaluated {
-		paramEvaluated = true
+			return nil, opErr
+		}
 
-		switch param := typeParam.(type) {
-		case *types.Document:
-			if !IsOperator(param) {
-				res = param
-				break
-			}
+		operator.operator = op
 
-			operator, err := NewOperator(param)
-			if err != nil {
-				var opErr OperatorError
-				if !errors.As(err, &opErr) {
-					return nil, lazyerrors.Error(err)
-				}
+	case *types.Array, float64, string, types.Binary, types.ObjectID, bool, time.Time,
+		types.NullType, types.Regex, int32, types.Timestamp, int64:
+		operator.param = param
 
-				if opErr.Code() == ErrInvalidExpression {
-					opErr.code = ErrInvalidNestedExpression
-				}
+	default:
+		panic(fmt.Sprint("wrong type of value: ", param))
+	}
 
-				return nil, opErr
-			}
+	return operator, nil
+}
 
-			if typeParam, err = operator.Process(doc); err != nil {
-				var opErr OperatorError
-				if !errors.As(err, &opErr) {
-					return nil, lazyerrors.Error(err)
-				}
+// Process implements Operator interface.
+func (t *typeOp) Process(doc *types.Document) (any, error) {
+	typeParam := t.param
 
-				return nil, err
-			}
+	if t.operator != nil {
+		var err error
+		typeParam, err = t.operator.Process(doc)
```

Comment:
Can we just return lazy error here? Seems a lot of code that isn't covered by tests. :thinking: